src/dbnode/client/session.go

```diff
@@ -1988,7 +1992,8 @@ func (s *session) FetchBlocksFromPeers(
 		complete = int64(0)
 		doneCh   = make(chan error, 1)
 		outputCh = make(chan peerBlocksDatapoint, 4096)
-		result   = newStreamBlocksResult(s.opts, opts, outputCh,
+		nsCtx    = namespace.NewContextFor(nsMetadata.ID(), s.opts.SchemaRegistry())
```

Comment:
Could this not also be `namespace.NewContextFrom(nsMetadata)`?

Reply:
yes